sw/host/opentitanlib/src/app/mod.rs

```diff
@@ -166,6 +166,8 @@ impl I2cConfiguration {
 
 pub struct TransportWrapperBuilder {
     interface: String,
+    provides_list: Vec<(String, String)>,
+    requires_list: Vec<(String, String)>,
```

Comment:
Is there a reason these cannot also be HashMaps?

Reply:
The `build()` method is the time when the code gets the final piece, the `provides_map` from the underlying transport, (if it is a proxy through a session daemon, which has itself loaded configuration files).  And that is the time when I currently call `consolidate_provides_map()` and `verify_requires_list`, producing errors in case there are any conflicting requirements.
If I were to incrementally combine configuration files into a `HashMap` here, then the verification would have to take place incrementally as well, I thought it would be easier to do it all in one go.

Reply:
Keeping lists is consistent with e.g. `pin_conf_list` records the separate aspects of a single pin, that could be declared in different files.  I realize that `pin_alias_map` is different, but I have come to think that it may be a mistake as `pin_alias_map.insert()` below could actually overwrite a previous declaration, meaning that the tool will not detect the same ping being declared first as alias of one thing, and later as alias of something else.  The "alias_of" probably should become another optional field of `PinConfiguration`, and `consolidate_pin_conf_map()` would have to check that there are not two `PinConfigurations` for the same pin, declaring different aliases.